wrapper/xgboost.py

```diff
@@ -478,10 +490,16 @@ def load_model(self, fname):
 
         Parameters
         ----------
-        fname : string
-            Input file name.
+        fname : string of file handle
+            Input file name or file handle object.
         """
-        xglib.XGBoosterLoadModel(self.handle, c_str(fname))
+        if isinstance(fname, string_types):  # assume file name
+            xglib.XGBoosterLoadModel(self.handle, c_str(fname))
+        else:
+            buf = fname.getbuffer()
+            length = ctypes.c_ulong(buf.nbytes)
+            ptr = ctypes.byref(ctypes.c_void_p.from_buffer(buf))
```

Comment:
Yeah what you're saying makes sense to me, though I couldn't figure out how to do it for `void_p` without the byref. I'll make your changes, since it seems more robust

The segfault is in `XGBoostGetModelRaw` so there's likely an issue in there I haven't spotted too.